v11y/index/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+            state: schema.add_text_field("state", STRING | FAST | STORED),
+
+            title: schema.add_text_field("title", TEXT | FAST | STORED),
+            description: schema.add_text_field("description", TEXT | FAST | STORED),
+
+            cvss31_score: schema.add_f64_field("cvss31_score", FAST | INDEXED | STORED),
+            cvss30_score: schema.add_f64_field("cvss30_score", FAST | INDEXED | STORED),
+        };
+        Self {
+            schema: schema.build(),
+            fields,
+        }
+    }
+
+    fn index_common(&self, document: &mut Document, metadata: &common::Metadata, container: &common::CnaContainer) {
+        document.add_text(self.fields.id, &metadata.id);
+    }
+
+    fn index_published_cve(&self, cve: &Published) -> Result<Document, SearchError> {
+        log::debug!("Indexing published CVE document");
+
+        let mut document = doc!();
+
+        document.add_text(self.fields.state, "published");
+        self.index_common(&mut document, &cve.metadata.common, &cve.containers.cna.common);
+
+        if let Some(title) = &cve.containers.cna.title {
+            document.add_text(self.fields.title, &title);
+        }
+
+        for desc in &cve.containers.cna.descriptions {
+            document.add_text(self.fields.description, &desc.value);
+        }
+
+        for metric in &cve.containers.cna.metrics {
+            if let Some(score) = metric.cvss_v3_1.as_ref().and_then(|v| v["vectorString"].as_str()) {
+                match Base::from_str(score) {
+                    Ok(score) => document.add_f64(self.fields.cvss31_score, score.score().value()),
+                    Err(err) => log::warn!("Failed to parse CVSS 3.1: {err}"),
+                }
+            } else if let Some(score) = metric.cvss_v3_0.as_ref().and_then(|v| v["vectorString"].as_str()) {
+                match Base::from_str(score) {
+                    Ok(score) => document.add_f64(self.fields.cvss30_score, score.score().value()),
+                    Err(err) => log::warn!("Failed to parse CVSS 3.0: {err}"),
+                }
+            }
+        }
+
+        log::debug!("Indexed {:?}", document);
+        Ok(document)
+    }
+
+    fn index_rejected_cve(&self, cve: &Rejected) -> Result<Document, SearchError> {
+        log::debug!("Indexing rejected CVE document");
+
+        let mut document = doc!();
+
+        document.add_text(self.fields.state, "rejected");
+        self.index_common(&mut document, &cve.metadata.common, &cve.containers.cna.common);
+
+        log::debug!("Indexed {:?}", document);
+        Ok(document)
+    }
+
+    fn resource2query(&self, resource: &Cves) -> Box<dyn Query> {
+        const PACKAGE_WEIGHT: f32 = 1.5;
+        const CREATED_WEIGHT: f32 = 1.25;
+        match resource {
+            Cves::Id(value) => Box::new(TermQuery::new(
+                Term::from_field_text(self.fields.id, value),
+                Default::default(),
```

Comment:
This will exactly match the ID, though I think you may want a string/substring matching that you get with create_string_query?